tap_pendo/streams.py

```diff
@@ -126,19 +139,24 @@ class Stream():
     key_properties = KEY_PROPERTIES
     stream = None
     period = None
+    request_retry_count = 1
     # initialized the endpoint attribute which can be overriden by child streams based on
     # the different parameters used by the stream.
     endpoint = Endpoints("/api/v1/aggregation", "POST")
 
     def __init__(self, config=None):
         self.config = config
 
-    def send_request_get_results(self, req, endpoint, params, count, **kwargs):
-        # Set request timeout to config param `request_timeout` value.
         # If value is 0,"0", "" or None then it will set default to default to 300.0 seconds if not passed in config.
         config_request_timeout = self.config.get('request_timeout')
-        request_timeout = config_request_timeout and float(config_request_timeout) or REQUEST_TIMEOUT # pylint: disable=consider-using-ternary
-        resp = session.send(req, timeout=request_timeout)
+        if config_request_timeout and float(config_request_timeout) > 0:
```

Comment:
It feels weird to have a float for a timeout, is there a good reason it's not an int?

Reply:
Although `int()` seems to be more appropriate, it seems that using `float()` is the common practice in other taps as well, and therefore, I believe we should maintain consistency with what we have been using.